File: Tutoriels/Python/POO/Heritage/src/GPerson.py
#================================================
class GPerson: 
    def __init__(self, name):
        self.m_name = name
        self.m_city = "Paris"
    #================================================
    def __str__(self):
        return "{0} | {1}".format(self.m_name, self.m_city)
#================================================

File: Tutoriels/Cpp/POO/Heriatge/src/GPerson.cpp
//===============================================
#include "GPerson.h"
//===============================================
GPerson::GPerson() {
    m_name = "";
    m_email = "";
}
//===============================================
GPerson::GPerson(const string& name, const string& email) {
    m_name = name;
    m_email = email;
}
//===============================================
GPerson::~GPerson() {

}
//===============================================
void GPerson::print() const {
    string lData = "";
    lData += "Personne: Je suis Mineur sans emploi";
    lData += "\n";
    cout << lData.c_str();
}
//===============================================
void GPerson::printInfo() const {
    string lData = "";
    lData += "Personne: Je m'appelle ";
    lData += m_name + " et mon email est " + m_email;
    lData += "\n";
    cout << lData.c_str();
}
//===============================================
